### ISSUE TITLE
 How can we get the class label at first when infer test-set?

### ISSUE THREAD

**Issue by J-Lam0** [Ruxie189/WSS_POLE#3]
As clip_selection.py shown, we need to use target class label to help us select the most suitable text prompt for images. So how can we get the class label at first when infer test-set? Did it show anywhere in code? Or can you explain this in more detail?

**Comment by Ruxie189** [comment]
Hii, so you clip_selection.py is to get class names for train and validation set. We use this class names to train the model to generate cams. The miou on cams thus reported are on train and val set. Once this is done you train the IR-Net (train_irn.py) with these class activation maps. Finally you need to use the trained ir net model  to generate pseduo labels for the test set (which does not require class text names and is standard in literature) and evaluate the model on the website. So the miou scores on test set are on segmentation masks and not the cams. Hope it helps!
